lib/src/component/error_boundary.dart

```diff
@@ -183,4 +250,136 @@ class ErrorBoundaryComponent<T extends ErrorBoundaryProps, S extends ErrorBounda
+  //           "times out", we should not just keep remounting since the component
+  //           should - at that point - be considered "unrecoverable",
+  //           and the repeated errors will most likely lock up the main thread.
+  //
+  //     [2.2] So we'll set `state.showFallbackUIOnError` to true, which will cause our default fallback UI
+  //           to be rendered - which is simply a String snapshot of the DOM (`_domAtTimeOfError`).
+  //
+  //           [2.2.1] We will also fire `props.onComponentIsUnrecoverable` if set - to give the
+  //                   consumer the ability to have knowledge of the unrecoverable error state - and
+  //                   to optionally manually recover from the error by:
+  //
+  //                   1. Passing in a new child (see: componentDidUpdate)
+  //                   2. Calling the `reset()` instance method
+  //
+  //           [2.2.2] Since we should __never__ throw an error from our... uh... error boundary,
+  //                   wrap in a try catch just in case `findDomNode` throws as a result of the
+  //                   wrapped react tree rendering a string instead of a composite or dom component.
+  // ---------------------------------------------- /\ ----------------------------------------------
+
+  String _domAtTimeOfError;
+  /*Error||Exception*/dynamic _lastError;
+  String _lastErrorInfo;
+  Timer _identicalErrorTimer;
+
+  /// Called by [componentDidCatch].
+  void _handleErrorInComponentTree(/*Error||Exception*/dynamic error, /*NativeJavascriptObject*/dynamic jsErrorInfo) {
+    bool sameErrorWasThrownTwiceConsecutively =
+        error.toString() == _lastError?.toString() && _getReadableErrorInfo(jsErrorInfo) == _lastErrorInfo;
+
+    // ----- [1] ----- //
+    if (props.fallbackUIRenderer != null) {
+      _lastError = error; // [1.1]
+      _lastErrorInfo = _getReadableErrorInfo(jsErrorInfo); // [1.1]
+
+      setState(newState()..hasError = true); // [1]
+      return;
+    }
+    // ----- [2] ----- //
+    else {
+      if (sameErrorWasThrownTwiceConsecutively) { // [2.1]
+        try { // [2.2.2]
+          _domAtTimeOfError = findDomNode(this)?.innerHtml; // [2.2]
+        } catch (_) {}
+
+        if (props.onComponentIsUnrecoverable != null) { // [2.2.1]
+          props.onComponentIsUnrecoverable(error, _getReadableErrorInfo(jsErrorInfo));
+        }
+      } else {
+        _lastError = error;
+        _lastErrorInfo = _getReadableErrorInfo(jsErrorInfo);
+      }
+
+      setState(newState()
+        ..hasError = true
+        ..showFallbackUIOnError = sameErrorWasThrownTwiceConsecutively // [2.2]
+      );
+
+      _startIdenticalErrorTimer();
+    }
+  }
+
+  // [2.2]
+  ReactElement _renderStringDomAfterUnrecoverableErrors(_, __) {
+    return (Dom.div()
+      ..key = 'ohnoes'
+      ..addTestId('ErrorBoundary.unrecoverableErrorInnerHtmlContainerNode')
+      ..['dangerouslySetInnerHTML'] = {'__html': _domAtTimeOfError} ?? ''
```

Comment:
What are the cases when the null check will return `''`? Because the `Map` is always set with a key, won't it always be non-null?

Reply:
Is this supposed to be on `_domAtTimeOfError`?
```suggestion
      ..['dangerouslySetInnerHTML'] = {'__html': _domAtTimeOfError ?? ''}
```

Maybe we should add a test where the child component is a or renders a fragment, which would represent a case where `_domAtTimeOfError` would be `null`:
```dart
ErrorBoundary([ // Can't use variadic children or the fragment will be interpreted as two separate children
  // A list-style "fragment" as the single child.
  [
    (Dom.div()..key = 1)(),
    (FaultyComponent()..key = 2)(),
  ],
])
```
Or actually, maybe even an empty component would do...

Reply:
Nice catch on this typo @joebingham-wk !!!

Reply:
Fixed the typo, added edge case test coverage in https://github.com/Workiva/over_react/pull/350/commits/a9e864eeef2abd045c977aa1377569d7fcb3d399